paddle/phi/api/yaml/ops.yaml

```diff
@@ -1686,6 +1686,18 @@
     func : matrix_power
   backward : matrix_power_grad
 
+- op : max_pool2d_v2
```

Comment:
`xxx_v2` isn't a good name for op, it should be same with python api.

Reply:
Under the current design framework, this is indeed a problem. We may need more information for making a decision：

- What is the accurate value of performance improvement for `max_pool2d_v2`? 
- Is `saved_idx` necessary? If not using `saved_idx`, the performance will decrease?

Reply:
1. The performance gain is huge, in NHWC layout and FP16 dtype. This figure shows the avg time of a pooling fwd + bwd: 
![image](https://github.com/PaddlePaddle/Paddle/assets/29906817/02fdc18b-28ec-4f50-8664-07c28f213ec0)

The speedup is at least 2.5x, up to 7.5x, depending on config.

2. Yes it is necessary for NHWC layout. The runtime fused pool kernel cannot work without it.

Reply:
> 2. Yes it is necessary for NHWC layout. The runtime fused pool kernel cannot work without it.

Is it a fused kernel?  It will be used from python api or pass?

Reply:
It is technically not a fused kernel -- it's only a max pooling and can be called from either python api or pass. However, the kernel is generated JIT using runtime fusion engine.

Reply:
Can `saved_idx` be add as a new output of `pool2d`？

Reply:
Yes, but it will involve a lot of changes -- adding an additional saved_idx to all forward / backward kernel functions of all backends (xpu for example, even though they will never use it); I also have to change the pooling trt converter to make sure TRT converter recognize this new signature. It could be 20+ file changes, for adding a new output. Compared to that, I think create a new op is a relatively lightweight solution.